components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -1548,6 +1548,59 @@ END SUBROUTINE access_lookup_table_coll
 
   !==========================================================================================!
 
+  real(rtype) function MurphyKoop_svp(t, i_type)
+
+    implicit none
+
+    !-------------------------------------------------------------------
+    ! Compute saturation vapor pressure (returned in units of pa)
+    ! Inputs:
+    ! "t", units [K]
+    !  i_type refers to saturation with respect to liquid (0) or ice (1)
+    !--------------------------------------------------------------------
+
+    !Murphy & Koop (2005)
+    real(rtype), intent(in) :: t
+    integer,intent(in)      :: i_type
+
+    !local vars
+    character(len=1000) :: err_msg
+    real(rtype)         :: logt
+
+    !parameters for ice saturation eqn
+    real(rtype), parameter :: ic(4)  =(/9.550426_rtype, 5723.265_rtype, 3.53068_rtype, &
+         0.00728332_rtype/)
+
+    !parameters for liq saturation eqn
+    real(rtype), parameter :: lq(10) = (/54.842763_rtype, 6763.22_rtype, 4.210_rtype, &
+         0.000367_rtype, 0.0415_rtype, 218.8_rtype, 53.878_rtype, 1331.22_rtype,       &
+         9.44523_rtype, 0.014025_rtype /)
+
+    logt = bfb_log(t)
+
+    if (i_type .eq. 1 .and. t .lt. zerodegc) then
+
+       !(good down to 110 K)
+       MurphyKoop_svp = exp(ic(1) - (ic(2) / t) + (ic(3) * logt) - (ic(4) * t))
+
+    elseif (i_type .eq. 0 .or. t .ge. zerodegc) then
+
+       ! (good for 123 < T < 332 K)
+       MurphyKoop_svp = exp(lq(1) - (lq(2) / t) - (lq(3) * logt) + (lq(4) * t) + &
+            (tanh(lq(5) * (t - lq(6))) * (lq(7) - (lq(8) / t) - &
+            (lq(9) * logt) + lq(10) * t)));
+    else
```

Comment:
Let's add an "else if (i_type.ne.0 .and. i_type.ne.1)" here and use "else" to handle only the T=NaN case.